### PR TITLE
DEPR: remove get_values, SparseArray.values

### PR DESCRIPTION
**Pull request by jbrockmendel**
The cleanups in libreduction are made possible by SparseArray.values going away.

### CODE REVIEW

**Comment by gfyoung** [comment]
I would have thought these changes would have been fine...just need to diagnose the CI failures.

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -1461,28 +1461,6 @@ def value_counts(self, dropna=True):
...
-        """
-        warn(
-            "The 'get_values' method is deprecated and will be removed in a "
-            "future version",
-            FutureWarning,
-            stacklevel=2,
-        )
-        return self._internal_get_values()
-
     def _internal_get_values(self):
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/categorical.py]: Can you add part of the above docstring to this method? (the interesting parts for internal use, eg what it returns)

(and potentially the same comment for some of the other cases below)

Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py]: good idea

**Comment by WillAyd** [comment]
I think there is some JSON code that can get cleaned up as part of this:

https://github.com/pandas-dev/pandas/blob/ed20822a54e3863b393554ecca801654af105555/pandas/_libs/src/ujson/python/objToJSON.c#L216

**Comment by jbrockmendel** [comment]
> I think there is some JSON code that can get cleaned up as part of this:

Nice.  Want me to do that here or save for follow-up?

**Comment by WillAyd** [comment]
Up to you

**Comment by jorisvandenbossche** [comment]
> > I think there is some JSON code that can get cleaned up as part of this:
>
> Nice. Want me to do that here or save for follow-up?

Let's do that here I would say, it's from the same PR that deprecated SparseArray.values: https://github.com/pandas-dev/pandas/pull/26421 (you are already removing part of the things of that PR that can be removed)

File: pandas/core/generic.py [hunk 2]
```diff
@@ -5500,7 +5497,7 @@ def get_values(self):
         0  1   True  1.0
         1  2  False  2.0
 
-        >>> df.get_values()
+        >>> df._internal_get_values()
```
Comment (jreback) [inline pandas/core/generic.py]: i would just remove these examples as they are no longer relevant

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. ping when ready (if addressing JSON things)

**Comment by jbrockmendel** [comment]
removed examples from docstring, left the json since it wasnt obvious what to do in the places where is_sparse_array is currently called

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by jorisvandenbossche** [comment]
>  left the json since it wasnt obvious what to do in the places where is_sparse_array is currently called

From looking at my old PR, I think you can just remove the `is_sparse_array` where it is used (so change it back to how it was before that PR). SparseArray won't have a `values` attribute anymore, so that check will work fine now (and not trigger a warning).

**Comment by jbrockmendel** [comment]
removed is_sparse_array, green

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**